app/services/bgs/dependent_v2_service.rb

```diff
@@ -74,6 +74,30 @@ def service
       @service ||= BGS::Services.new(external_uid: icn, external_key:)
     end
 
+    def log_bgs_errors(error)
+      if Flipper.enabled?(:va_dependents_bgs_extra_error_logging)
+        error_messages = ['HTTP error (302)', 'HTTP error (500)', 'HTTP error (502)', 'HTTP error (504)']
+        error_type_map = {
+          'HTTP error (302)' => '302',
+          'HTTP error (500)' => '500',
+          'HTTP error (502)' => '502',
+          'HTTP error (504)' => '504'
+        }
+        nested_error_message = error.cause&.message
+
+        error_type = error_messages.select do |et|
```

Comment:
@TaiWilkin if you know of a way to check the actual status code, I think that would be more concise then this approach

Reply:
Based on the BGS-ext cassettes, I think you should be able to do request.status or request.status.code, but I'm not certain.